File: Jan2021/05Jan/TwoSum.java
/*
Problem Name: 1. Two Sum
Problem Link: https://leetcode.com/problems/two-sum/
*/
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> map = new HashMap<Integer,Integer>();
        int n = nums.length;
        for(int i=0;i<n;i++){
            if(map.containsKey(target - nums[i]))
                return new int[]{i,map.get(target - nums[i])};
            map.put(nums[i],i);
        }
        return new int[]{};
    }
}

File: Jan2021/05Jan/TwoSum.cpp
/*
Problem name: 1. Two Sum
Problem link: https://leetcode.com/problems/two-sum/
*/

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> indices;
        map<int, int> mp;
        for(int i=0;i<nums.size();++i){
            if(mp.count(target-nums[i]) && mp[target-nums[i]]!=i){
                indices.push_back(mp[target-nums[i]]);
                indices.push_back(i);
                break;
            }
            mp[nums[i]]=i;
        }
        return indices;
    }
};
